be/src/exec/pipeline/scan/scan_operator.cpp

```diff
@@ -443,7 +443,11 @@ Status ScanOperator::_pickup_morsel(RuntimeState* state, int chunk_source_index)
     return Status::OK();
 }
 
-void ScanOperator::_merge_chunk_source_profiles() {
+void ScanOperator::_merge_chunk_source_profiles(RuntimeState* state) {
+    auto query_ctx = state->exec_env()->query_context_mgr()->get(state->query_id());
```

Comment:
`query_context_mgr()->get` returns `shared_ptr`, so maybe `const auto& query_ctx` to avoid copy `shared_ptr`?

Reply:
auto& may reference to `get`'s local variable?